packages/rpc-core/src/unix-timestamp.ts

```diff
@@ -0,0 +1,14 @@
+export type UnixTimestamp = number & { readonly __unixTimestamp: unique symbol };
+
+export function assertIsUnixTimestamp(putativeTimestamp: number): asserts putativeTimestamp is UnixTimestamp {
+    // see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date#the_epoch_timestamps_and_invalid_date
+    try {
+        if (putativeTimestamp > 8.64e15) {
```

Comment:
Tiny nit: both directions. And add a test.

`if (putativeTimestamp > 8.64e15 || putativeTimestamp < -8.64e15) {`